--- qsslsocket_openssl.cpp_orig	2018-10-30 23:42:02.594549970 +0300
+++ qsslsocket_openssl.cpp	2018-10-30 23:42:29.006532151 +0300
@@ -747,8 +747,12 @@
                 certFiles.insert(it.fileInfo().canonicalFilePath());
             }
         }
+    #ifdef Q_OS_WEBOS
+        systemCerts.append(QSslCertificate::fromPath(QLatin1String("/media/cryptofs/apps/usr/palm/applications/com.nizovn.cacert/files/cert.pem"), QSsl::Pem));
+    #else
         for (const QString& file : qAsConst(certFiles))
             systemCerts.append(QSslCertificate::fromPath(file, platformEncodingFormat));
+    #endif
 # ifndef Q_OS_ANDROID
         systemCerts.append(QSslCertificate::fromPath(QLatin1String("/etc/pki/tls/certs/ca-bundle.crt"), QSsl::Pem)); // Fedora, Mandriva
         systemCerts.append(QSslCertificate::fromPath(QLatin1String("/usr/local/share/certs/ca-root-nss.crt"), QSsl::Pem)); // FreeBSD's ca_root_nss
